lib/nimble_options/validation_error.ex

```diff
@@ -40,4 +41,31 @@ defmodule NimbleOptions.ValidationError do
 
     message <> suffix
```

Comment:
Wouldn't it be really important to also redact when turning the exception into a message? That's what happens when you `raise` said exception so I think this is what we want, not `Inspect` (as you generally won't inspect the exception).

Reply:
The redaction happens when the struct is built so the message should be good at this point. Unless there's something I'm missing.

I think redacting on `Inspect` is important. I would be surprised if I was using this feature and some secrets leaked due to inspecting `ValidationError`. If we're going to redact it should be everywhere.

Reply:
Added a test showing message is redacted when using `NimbleOptions.validate!/2`.

Reply:
@whatyouhide is there anything more here that needs done?